code/__DEFINES/skills.dm

```diff
@@ -24,120 +24,200 @@
+#define SKILL_FIREARMS_TRAINED 2
 
 ///Damage mod for having the weapon specific skill above 0
 #define FIREARM_SKILL_DAM_MOD 0.05
 
-//pistols skill
-//increase or decrase accuracy, recoil, and firing delay of pistols and revolvers.
-#define SKILL_PISTOLS_DEFAULT 0	//marines
-#define SKILL_PISTOLS_TRAINED 1	//special training
-
-//smgs skill
-//increase or decrase accuracy, recoil, and firing delay of submachineguns.
-#define SKILL_SMGS_DEFAULT 0	//marines
-#define SKILL_SMGS_TRAINED 1	//special training
-
-//rifles skill
-//increase or decrase accuracy, recoil, and firing delay of rifles.
-#define SKILL_RIFLES_DEFAULT 0	//marines
-#define SKILL_RIFLES_TRAINED 1	//special training
-
-//shotguns skill
-//increase or decrase accuracy, recoil, and firing delay of shotguns.
-#define SKILL_SHOTGUNS_DEFAULT 0	//marines
-#define SKILL_SHOTGUNS_TRAINED 1	//special training
-
-//heavy weapons skill
-//increase or decrase accuracy, recoil, and firing delay of heavy weapons (non spec weapons, e.g. flamethrower).
-#define SKILL_HEAVY_WEAPONS_DEFAULT 0	//marines
-#define SKILL_HEAVY_WEAPONS_TRAINED 1	//special training
-
-//smartgun skill
-//increase or decrase accuracy, recoil, and firing delay for smartgun, and whether we can use smartguns at all.
-#define SKILL_SMART_DEFAULT -4 //big negative so the effects are far worse than pistol/rifle untrained
-#define SKILL_SMART_USE -3 //can use smartgun
-#define SKILL_SMART_TRAINED 0 //default for smartgunner
+/**
+ * Pistols skill; Increase or decrase accuracy, recoil, and firing delay of pistols and revolvers.
+ * DEFAULT = Marines
+ * TRAINED = Special training (Synths for tasers; ERT)
+ */
+#define SKILL_PISTOLS_DEFAULT 0
+#define SKILL_PISTOLS_TRAINED 1
+
+/**
+ * SMG skill; Increase or decrase accuracy, recoil, and firing delay of SMGs.
+ * DEFAULT = Marines
+ * TRAINED = Special training (ERT)
+ */
+#define SKILL_SMGS_DEFAULT 0
+#define SKILL_SMGS_TRAINED 1
+
+/**
+ * Rifles skill; Increase or decrase accuracy, recoil, and firing delay of rifles.
+ * DEFAULT = Marines
+ * TRAINED = Special training (ERT)
+ */
+#define SKILL_RIFLES_DEFAULT 0
+#define SKILL_RIFLES_TRAINED 1
+
+/**
+ * Shotgun skill; Increase or decrase accuracy, recoil, and firing delay of shotguns.
+ * DEFAULT = Marines
+ * TRAINED = Special training (ERT)
+ */
+#define SKILL_SHOTGUNS_DEFAULT 0
+#define SKILL_SHOTGUNS_TRAINED 1
+
+/**
+ * Heavy weapons skill; Increase or decrase accuracy, recoil, and firing delay of heavy weapons (eg. machineguns, pulse rifles, flamethrower).
+ * DEFAULT = Marines
+ * TRAINED = Special training (ERT)
+ */
+#define SKILL_HEAVY_WEAPONS_DEFAULT 0
+#define SKILL_HEAVY_WEAPONS_TRAINED 1
+
+/**
+ * Smartgun skill; Increase or decrase accuracy, recoil, and firing delay for smartgun, and whether we can use smartguns at all.
+ * DEFAULT = Big negative so the effects are far worse than pistol/rifle untrained
+ * USE = Can use a smartgun (CO)
+ * TRAINED = Smartgunners
+ * EXPERT = Unused
+ * MASTER = Deathsquad
+ */
```

Comment:
oh yeah, you should probably these uh, accurate.
Increasing skill improves acc, scatter and damage. It does not improve firing speed (although for heavy I think it improved reload speed?

Reply:
in gun_system.dm, lines 1736-1751:

```/obj/item/weapon/gun/proc/gun_on_cooldown(mob/user)
	var/added_delay = fire_delay
	if(user)
		if(user.skills.getRating(SKILL_FIREARMS) < SKILL_FIREARMS_DEFAULT)
			added_delay += 3 //untrained humans fire more slowly.
		else
			switch(gun_skill_category)
				if(SKILL_HEAVY_WEAPONS)
					if(fire_delay > 1 SECONDS) //long delay to fire
						added_delay = max(fire_delay - 3 * user.skills.getRating(gun_skill_category), 6)
				if(SKILL_SMARTGUN)
					if(user.skills.getRating(gun_skill_category) < 0)
						added_delay -= 2 * user.skills.getRating(gun_skill_category)
	var/delay = last_fired + added_delay
	if(gun_firemode == GUN_FIREMODE_BURSTFIRE)
		delay += extra_delay```